L1Trigger/L1TCaloLayer1/src/L1TCaloLayer1FetchLUTs.cc

```diff
@@ -122,8 +124,22 @@ bool L1TCaloLayer1FetchLUTs(
     return false;
   }
 
+  // HCAL FB LUT will be a 1*28 array:
+  //   ieta = 28 eta scale factors (1 .. 28)
+  //   So, index = ieta
+  auto fbLUTUpper = caloParams.layer1HCalFBLUTUpper();
+  auto fbLUTLower = caloParams.layer1HCalFBLUTLower();
+  if (fbLUTUpper.size() != 28) {
+    edm::LogError("L1TCaloLayer1FetchLUTs") << "caloParams.layer1HCalFBLUTUpper().size() != 28 !!";
+    return false;
+  }
+  if (fbLUTLower.size() != 28) {
+    edm::LogError("L1TCaloLayer1FetchLUTs") << "caloParams.layer1HCalFBLUTLower().size() != 28 !!";
+    return false;
+  }
+
   // Sanity check scale factors exist
-  if (useCalib && (ecalSF.empty() || hcalSF.empty() || hfSF.empty())) {
+  if (useCalib && (ecalSF.empty() || hcalSF.empty() || hfSF.empty() || fbLUTUpper.empty() || fbLUTLower.empty())) {
```

Comment:
Isn't this already included in the checks above?

Reply:
Technically yes, but this check is useful to distinguish whether the LUT size is off by a few entries or if it is missing from the caloParams file completely.